src/libsyntax/parse/diagnostics.rs

```diff
@@ -1205,4 +1232,24 @@ impl<'a> Parser<'a> {
         err.span_label(span, "expected expression");
         err
     }
+
+    /// Replace duplicated recovered arguments with `_` pattern to avoid unecessary errors.
```

Comment:
It's not really clear from this comment why this is necessary (e.g. what sort of errors this is preventing). It would be nice to expand this comment with an example or more of a description.

Reply:
Will do. I'll transcribe/reword [the convo with @petrochenkov](https://github.com/rust-lang/rust/pull/61331#discussion_r288891419) into this comment.

Reply:
Done.

Reply:
Great, thanks!